src/server/auth/server/api_server.go

```diff
@@ -657,16 +657,53 @@ func (a *apiServer) DeleteRoleBindingInTransaction(txnCtx *txnenv.TransactionCon
 	return nil
 }
 
-func bindingFromRequest(req *auth.ModifyRoleBindingRequest) *auth.Roles {
-	if len(req.Roles) == 0 {
-		return nil
+// rolesFromRoleSlice converts a slice of strings into *auth.Roles,
```

Comment:
Small refactor: validate the roles when we're converting from a slice of strings to an `auth.Roles` since we need to do it in Create and Modify.